src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -2062,68 +2145,141 @@ bool QgsMapToolEditMeshFrame::isSelectionGrapped( QgsPointXY &grappedPoint )
+    {
+      const QgsPointXY currentPoint =   mapVertexXY( mCurrentVertexIndex );
+      mForceByLineRubberBand->addPoint( currentPoint );
+      mCadDockWidget->setZ( QString::number( mapVertex( mCurrentVertexIndex ).z(), 'f' ), QgsAdvancedDigitizingDockWidget::WidgetSetMode::TextEdited );
+      mCadDockWidget->setPoints( QList < QgsPointXY>() << currentPoint << currentPoint );
+    }
+    else
+    {
+      mForceByLineRubberBand->addPoint( mapPoint );
+    }
 
-  for ( QgsMapLayer *layer : layers )
+    mForcingLineZValue.append( currentZValue() );
+  }
+  else if ( e->button() == Qt::RightButton )
   {
-    QgsVectorLayer *vectoLayer = qobject_cast<QgsVectorLayer *>( layer );
+    if ( mForceByLineRubberBand->numberOfVertices() == 0 )
+    {
+      forceByLineBySelectedFeature( e );
+      return;
+    }
+
+    QVector<QgsPoint> points;
+    QgsPolylineXY rubbergandLines = mForceByLineRubberBand->asGeometry().asPolyline();
+    double defaultValue = currentZValue();
+
+    if ( std::isnan( defaultValue ) )
+      defaultValue = defaultZValue();
 
-    if ( vectoLayer )
+    for ( int i = 0; i < rubbergandLines.count() - 1; ++i )
     {
-      const QgsFeatureList &features = vectoLayer->selectedFeatures();
+      points.append( QgsPoint( rubbergandLines.at( i ).x(),
+                               rubbergandLines.at( i ).y(),
+                               mForcingLineZValue.isEmpty() ? defaultValue : mForcingLineZValue.at( i ) ) );
+    }
+    std::unique_ptr<QgsLineString> forcingLine = std::make_unique<QgsLineString>( points );
+    forceByLine( QgsGeometry( forcingLine.release() ) );
+    mForceByLineRubberBand->reset( QgsWkbTypes::LineGeometry );
+    mForcingLineZValue.clear();
+  }
+}
 
-      QgsCoordinateTransform transform = canvas()->mapSettings().layerTransform( vectoLayer );
-      for ( const QgsFeature &feat : features )
+void QgsMapToolEditMeshFrame::forceByLineBySelectedFeature( QgsMapMouseEvent *e )
+{
+  QList<QgsMapToolIdentify::IdentifyResult> results;
+  const QMap< QString, QString > derivedAttributes;
+
+  QgsPointXY mapPoint = e->mapPoint();
+  double x = mapPoint.x(), y = mapPoint.y();
+  const double sr = QgsMapTool::searchRadiusMU( mCanvas );
+
+  const QList<QgsMapLayer *> layers = mCanvas->layers();
+  for ( QgsMapLayer *layer : layers )
+  {
+    if ( layer->type() == QgsMapLayerType::VectorLayer )
+    {
+      QgsVectorLayer *vectorLayer = static_cast<QgsVectorLayer *>( layer );
+      if ( vectorLayer->geometryType() == QgsWkbTypes::PolygonGeometry ||
+           vectorLayer->geometryType() == QgsWkbTypes::LineGeometry )
       {
-        QgsGeometry geom = feat.geometry();
+        QgsRectangle rect( x - sr, y - sr, x + sr, y + sr );
+        QgsCoordinateTransform transform = mCanvas->mapSettings().layerTransform( vectorLayer );
 
         try
         {
-          geom.transform( transform );
+          rect = transform.transformBoundingBox( rect, QgsCoordinateTransform::ReverseTransform );
```

Comment:
@PeterPetrik FYI this broke the builds on master.

Reply:
I have seen you fixed it? Thanks for that... not sure since there was all green from CI

Reply:
No problem, it happened to me too to see all green and then have a failure... not sure why it happens.

Reply:
not sure, maybe some change on the master in the meantime, but still without merge conflicts?

Reply:
that's possible, if ReverseTransform was moved to qgis.h and renamed.